fix: change wording for spec creation

Manuel-Suarez-Abascal opened the pull request:
<!-- Thanks for contributing! PLEASE...
- Read our contributing guidelines: https://github.com/cypress-io/cypress/blob/develop/CONTRIBUTING.md 
- Read our Code Review Checklist on coding standards and what needs to be done before a PR can be merged: https://github.com/cypress-io/cypress/blob/develop/CONTRIBUTING.md#Code-Review-Checklist
- Mark this PR as "Draft" if it is not ready for review.
- Make sure you set the correct base branch based on what packages you're changing: https://github.com/cypress-io/cypress/blob/develop/CONTRIBUTING.md#branches
-->

- Closes #21908

### User-facing changelog

Fix misleading wording when creating a new starter spec file in the UI.

### Additional details

N/A

### Steps to test

- Launch Cypress's test runner.
- Click on the button `New Spec` button.
- You should see the new wording in the cards to create a new starter spec file.

### How has the user experience changed?

### PR Tasks


- [X] Have tests been added/updated?
- [ ] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? <!-- Link to PR here -->
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

Manuel-Suarez-Abascal committed:
fix: change wording for spec creation

Manuel-Suarez-Abascal committed:
fix: refactor test files

lmiller1990 commented:
Did you apply the requested changes from https://github.com/cypress-io/cypress/pull/24825 @Manuel-Suarez-Abascal?

Thanks, I triggered a CI run for you.

Manuel-Suarez-Abascal commented:
> Did you apply the requested changes from #24825 @Manuel-Suarez-Abascal?
> 
> Thanks, I triggered a CI run for you.

👋🏻 
I have replied to Emily [here](https://github.com/cypress-io/cypress/pull/24825/files/cb11c3c3d520451f71dc5aadb59068dd41105f3a#r1056628681). Waiting for confirmation on her end.

emilyrohrbough commented:
@Manuel-Suarez-Abascal Thanks for linking to your comment. I likely wouldn't have looked at the closed review. Glad you were able to get the CLA issue figured out!

emilyrohrbough commented on packages/frontend-shared/src/locales/en-US.json:
```diff
@@ -111,8 +111,8 @@
       "importEmptySpec": {
-        "header": "Create a new spec",
-        "description": "We'll generate an empty spec file to start testing components.",
+        "header": "Create starter spec",
```
I think we should keep the header as `Create new spec`  and just update the description to use the updated term of starter spec. I don't have a super strong opinion on starter vs template -- I've just never heard the term starter before which is why I suggested template.

Manuel-Suarez-Abascal replied:
Sounds good, I'll make the changes suggested in the closed PR's comments & ping you guys when I'm done :)

Manuel-Suarez-Abascal committed:
fix: address PR comments

Manuel-Suarez-Abascal committed:
fix: address PR comments

Manuel-Suarez-Abascal commented:
@lmiller1990 @emilyrohrbough I have addressed the PR comment. Let me know your thoughts or any other changes that I might have missed.

emilyrohrbough commented:
@Manuel-Suarez-Abascal looks like there a few failing tests with these changes.

Manuel-Suarez-Abascal committed:
Merge branch 'develop' into issue-21908

Manuel-Suarez-Abascal committed:
Merge branch 'develop' into issue-21908

lmiller1990 commented:
One of the component tests looks to be failing. Take a quick look @Manuel-Suarez-Abascal?

Thanks! Just do `cd packages/app && yarn cypress:run:ct --chrome` locally if you need to verify they all pass before pushing, they are pretty quick. Throw in `--headed` to watch them.

Manuel-Suarez-Abascal commented:
> One of the component tests looks to be failing. Take a quick look @Manuel-Suarez-Abascal?
> 
> Thanks! Just do `cd packages/app && yarn cypress:run:ct --chrome` locally if you need to verify they all pass before pushing, they are pretty quick. Throw in `--headed` to watch them.

Thanks, @lmiller1990 I have to try to fix the error, but I'm not exactly sure why it's failing. I will need more time debugging it, but it doesn't find the `Create new spec` text on the screen which as you see in the screenshot below it's present. Thoughts 💭 ?

<img width="1344" alt="Screenshot 2022-12-29 at 8 05 08 AM" src="https://user-images.githubusercontent.com/32891808/209955262-3a9faf5f-7247-4219-b480-3d6b4f54e328.png">

Manuel-Suarez-Abascal committed:
Merge branch 'develop' into issue-21908

Commit:
Merge branch 'develop' into issue-21908

Manuel-Suarez-Abascal committed:
fix: address PR comment & fix CI failures

Manuel-Suarez-Abascal committed:
Merge remote-tracking branch 'origin/issue-21908' into issue-21908

Manuel-Suarez-Abascal commented:
@emilyrohrbough @lmiller1990 I have addressed the PR comments & fixed the failing spec files. I think it's ready for another review :)

<img width="1212" alt="Screenshot 2022-12-30 at 11 01 34 AM" src="https://user-images.githubusercontent.com/32891808/210089544-9fcae000-3519-4b71-8638-5e56ba467893.png">

emilyrohrbough committed:
fix tests & update verbiage in a few other places
